### PR TITLE
Remove deprecated global state from specs

### PR DESCRIPTION
**Pull request by marcospereira**
## Purpose

Removes uses of deprecated APIs that are still accessing global state. While this is only specs code, it is better that we can have the non-deprecated access better tested.

### CODE REVIEW

File: framework/src/play-integration-test/src/test/scala/play/it/http/parsing/JsonBodyParserSpec.scala [hunk 1]
```diff
@@ -50,41 +55,44 @@ class JsonBodyParserSpec extends PlaySpecification {
...
       import scala.concurrent.ExecutionContext.Implicits.global
 
-      val fooParser = BodyParsers.parse.json.validate {
+      implicit val fooParser: BodyParser[Foo] = jsonBodyParser.validate {
         _.validate[Foo].asEither.left.map(e => BadRequest(JsError.toJson(e)))
       }
-      parse("""{"a":1,"b":"bar"}""", Some("application/json"), "utf-8", fooParser) must beRight
-      parse("""{"foo":"bar"}""", Some("application/json"), "utf-8", fooParser) must beLeft
-      parse("""{"a":1}""", Some("application/json"), "utf-8", fooParser) must beLeft
+      parse("""{"a":1,"b":"bar"}""", Some("application/json"), "utf-8")(app.materializer, fooParser) must beRight
```
Comment (gmethvin) [inline framework/src/play-integration-test/src/test/scala/play/it/http/parsing/JsonBodyParserSpec.scala]: Couldn't you pass those parameters implicitly?

File: framework/src/play-ahc-ws/src/test/scala/play/libs/ws/WSSpec.scala [hunk 2]
```diff
@@ -3,35 +3,49 @@
...
-    "uploads the stream" in new WithServer(app = uploadApp, port = 3333) {
-      val wsClient = app.injector.instanceOf(classOf[WSClient])
+        mat = components.materializer
 
-      val input = this.getClass.getClassLoader.getResourceAsStream("play/libs/ws/play_full_color.png")
-      val rep = wsClient.url("http://localhost:3333").post(input).toCompletableFuture.get()
+        import components.{ defaultActionBuilder => Action }
+        import play.api.routing.sird.{ POST => SirdPost, _ }
+        {
+          case SirdPost(p"/") => Action { req: Request[AnyContent] =>
```
Comment (gmethvin) [inline framework/src/play-ahc-ws/src/test/scala/play/libs/ws/WSSpec.scala]: Why did you rename to `SirdPost`?

File: framework/src/play-ahc-ws/src/test/scala/play/api/libs/ws/ahc/AhcWSSpec.scala [hunk 3]
```diff
@@ -354,6 +354,8 @@ class AhcWSSpec(implicit ee: ExecutionEnv) extends Specification with Mockito wi
     import java.io._
     import java.util.zip._
 
+    lazy val Action = new ActionBuilder.IgnoringBody()(_root_.controllers.Execution.trampoline)
```
Comment (gmethvin) [inline framework/src/play-ahc-ws/src/test/scala/play/api/libs/ws/ahc/AhcWSSpec.scala]: You can actually use `ActionBuilder.ignoringBody`

File: framework/src/play-ahc-ws/src/test/scala/play/libs/ws/WSSpec.scala (hunk 2 above)
Comment (marcospereira) [inline framework/src/play-ahc-ws/src/test/scala/play/libs/ws/WSSpec.scala]: Because `PlaySpecification` adds a `GET`, `POST`, etc from `HttpVerbs`.

File: framework/src/play-integration-test/src/test/scala/play/it/http/parsing/JsonBodyParserSpec.scala (hunk 1 above)
Comment (marcospereira) [inline framework/src/play-integration-test/src/test/scala/play/it/http/parsing/JsonBodyParserSpec.scala]: Yep. But we already have an implicit parser defined above. So I prefer to explicitly pass the parameter here to make the test clearer.

File: framework/src/play-ahc-ws/src/test/scala/play/api/libs/ws/ahc/AhcWSSpec.scala (hunk 3 above)
Comment (marcospereira) [inline framework/src/play-ahc-ws/src/test/scala/play/api/libs/ws/ahc/AhcWSSpec.scala]: Good catch. :-)

Done.
